### ISSUE TITLE
[Feature Request]: Marigold depth - high-quality diffusion-based monocular depth estimation

### ISSUE THREAD

**Issue by toshas** [Mikubill/sd-webui-controlnet#2330]
Improved monocular depth estimation algorithm based on fine-tuned stable diffusion. Expecting better guidance for depth-to-image and other models employing monocular depth.

https://marigoldmonodepth.github.io/
https://twitter.com/AntonObukhov1/status/1732946419663667464

**Comment by sdbds** [comment]
It looks like it could be combined with [Patchfusion](https://github.com/zhyever/PatchFusion), according to their paper, is a high-definition approach to depth maps that is technically independent of the base model.

**Comment by toshas** [comment]
Great; Marigold produces depth maps of the same revolution as input

**Comment by huchenlei** [comment]
Wow, Marigold inference is using a diffusion pipeline to produce depth maps? I think this is heavy enough that we make it an optional component. (Or we want a component management system for ControlNet?)

**Comment by toshas** [comment]
Yes, but it can be integrated similarly to https://github.com/kijai/ComfyUI-Marigold by copying the code, so there is no need for extra dependencies. When we roll out an update for Marigold, we will notify dependency repositories about improvements upstream!

**Comment by sdbds** [comment]
> Yes, but it can be integrated similarly to https://github.com/kijai/ComfyUI-Marigold by copying the code, so there is no need for extra dependencies. When we roll out an update for Marigold, we will notify dependency repositories about improvements upstream!

Apologies for the late reply, because while huggingface's range of packages are very good to replicate and implement, the iteration between versions makes breaking changes difficult to maintain. 
For example diffusers 0.20 and 0.21 are not common between them, nor are 0.24 and 0.25. 
We try not to use diffusers in webui, and we're trying to figure out how to read the diffusers config file to replicate the neural network constructs.

**Comment by sdbds** [comment]
> Wow, Marigold inference is using a diffusion pipeline to produce depth maps? I think this is heavy enough that we make it an optional component. (Or we want a component management system for ControlNet?)

I think a generic config neural network structure for reading diffusers might be the way to go.

**Comment by huchenlei** [comment]
Supported in https://github.com/lllyasviel/stable-diffusion-webui-forge
